### DISCUSSION TITLE
Why text padding is lost after setting active selection?

### DISCUSSION THREAD

**Discussion by marjune163** [fabricjs/fabric.js#9861] [General]
example:
https://codepen.io/marjune163/pen/vYwBBqO

```javascript
var canvas = this.__canvas = new fabric.Canvas('c');

var text =new fabric.IText("foo", {
  fontSize: 32,
  left: 50,
  top: 50,
  padding: 20,
  fill: "black",
});

canvas.add(text);
// controls are outside padding
canvas.setActiveObject(text);
canvas.renderAll();

btn.onclick=()=>{
  const activeObjects = canvas.getObjects();
  const selection = new fabric.ActiveSelection(activeObjects, { canvas });
  // controls are inside padding
  canvas.setActiveObject(selection);
  canvas.renderAll()
}

```

At first, a text selection with padding exists.
After clicking the button, it gets an selection object, and make it active. The padding is lost and controls is around text without padding.

Is there any issue about the code and any idea how to solve it?(make select by API and keep the padding?)

**Comment by asturur** [comment]
Every object has its own padding, including the active selection.
If you want to have everything with padding you have to set it as a default.

you can dow FabricObject.ownDefaults.padding = 20;
